src/config/ConfigManager.cpp

```diff
@@ -1172,6 +1173,9 @@ void CConfigManager::handleWorkspaceRules(const std::string& command, const std:
         rules                  = value.substr(WORKSPACE_DELIM + 1);
     }
 
+    const static std::string ruleOnCreatedEmtpy = "on-created-empty:";
```

Comment:
why?

Reply:
Just thought using a magic number seemed pointless. There should be no performace penalty on using a const static string, and using it does mean we can use the compiler to count its length and avoid the need of reviewers to have to double check string sizes. As a rule of thumb, I try to avoid magic numbers as much as possible, even if used only once.

With that being said, if you really think this doesn't belong, I'll be glad to remove it. In the meanwhile, I'll work on addressing your other comments in the next commit.